include/lmms_math.h

```diff
@@ -112,10 +123,11 @@ inline double fastPow(double a, double b)
 }
 
 
-//! returns 1.0f if val >= 0.0f, -1.0 else
-inline float sign(float val) 
+//! returns 1.0 if val >= 0.0, -1.0 else
+template<typename T>
+constexpr T sign(T val) noexcept
 { 
-	return val >= 0.0f ? 1.0f : -1.0f; 
+	return val >= 0.0 ? 1.0 : -1.0; 
```

Comment:
```suggestion
	return val >= 0 ? 1 : -1; 
```
I think this should work for both floating point types and signed integers, and should prevent a potential narrowing conversion warning when `T` is `float`.